transform-plugins/src/main/java/io/cdap/plugin/JSONParser.java

```diff
@@ -88,6 +91,25 @@ public void configurePipeline(PipelineConfigurer pipelineConfigurer) {
     extractMappings(collector);
   }
 
+  @Override
+  public void prepareRun(StageSubmitterContext context) throws Exception {
+    super.prepareRun(context);
+    extractMappings(context.getFailureCollector());
```

Comment:
This plugin has two mode: if `isSimple`, then the field lineage will be `config.field` to all the output fields. If not, then it is your following logic. Please add the first half logic